vnext/Microsoft.ReactNative.Cxx.UnitTests/ReactContextTest.cpp

```diff
@@ -20,18 +20,30 @@ struct ReactContextStub : implements<ReactContextStub, IReactContext> {
   }
 
   void CallJSFunction(
-      hstring const & /*moduleName*/,
-      hstring const & /*functionName*/,
-      JSValueArgWriter const & /*paramsArgWriter*/) noexcept {
-    VerifyElseCrashSz(false, "Not implemented");
+      hstring const &moduleName,
+      hstring const &functionName,
+      JSValueArgWriter const &paramsArgWriter) noexcept {
+    Module = moduleName;
+    Method = functionName;
+    auto writer = MakeJSValueTreeWriter();
+    paramsArgWriter(writer);
+    Args = TakeJSValue(writer);
   }
 
   void EmitJSEvent(
-      hstring const & /*eventEmitterName*/,
-      hstring const & /*eventName*/,
-      JSValueArgWriter const & /*paramsArgWriter*/) noexcept {
-    VerifyElseCrashSz(false, "Not implemented");
+      hstring const &eventEmitterName,
+      hstring const &eventName,
+      JSValueArgWriter const &paramsArgWriter) noexcept {
+    Module = eventEmitterName;
+    Method = eventName;
+    auto writer = MakeJSValueTreeWriter();
+    paramsArgWriter(writer);
+    Args = TakeJSValue(writer);
   }
+
+  std::wstring Module;
+  std::wstring Method;
```

Comment:
do these need to be wstring? seems like we could get away with 8 bit string

Reply:
In this case it is just a mock implementation for a test. The real code sends this data though the ABI-safe API that uses hstring which is based on 16-bit characters. Thus, using 8-bit characters is not practical because we would be required to convert strings from 8-bit to 16-bit every time we call the ABI-safe API.